### ISSUE TITLE
Passing in dictionary to np.log10 raises AttributeError

### ISSUE THREAD

**Issue by madisonmay** [numpy/numpy#6087]
```
Python 2.7.6 (default, Mar 22 2014, 22:59:56) 
[GCC 4.8.2] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import numpy as np
>>> np.log10({'key': 'value'})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: log10
```

I would expect appreciate a `TypeError` here instead, as the current error message is rather unintuitive.

**Comment by madisonmay** [comment]
Other relevant info:

```
$ pip freeze | grep numpy
numpy==1.9.2
```

**Comment by argriffing** [comment]
related https://github.com/numpy/numpy/issues/1611#issuecomment-19076282

**Comment by madisonmay** [comment]
Thanks for the link, glad to hear it's already an open issue.

**Comment by argriffing** [comment]
> glad to hear it's already an open issue.

I'm not sure this issue is more active now than it was a few years ago, but your help would be welcome!

**Comment by eric-wieser** [comment]
Closed in #12700
